execution_engine/src/core/engine_state/mod.rs

```diff
@@ -1679,23 +1680,29 @@ where
             }
         };
 
+        // the proposer of the block this deploy is in receives the gas from this deploy execution
+        let proposer_purse = {
+            let proposer_account: Account = match tracking_copy
+                .borrow_mut()
+                .get_account(correlation_id, proposer.into())
+            {
+                Ok(account) => account,
+                Err(error) => {
+                    return Ok(ExecutionResult::precondition_failure(error.into()));
+                }
+            };
```

Comment:
Maybe more concise:
```rust
            let proposer_account: Account = match tracking_copy
                .borrow_mut()
                .get_account(correlation_id, proposer.into())
                .map_err(|error| ExecutionResult::precondition_failure(error.into()))?;
```
(Also `proposer_main_purse_balance_key` below.)

Reply:
Did you notice the odd return semantics of exec?